File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/GremlinTsundere_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";


export class GremlinTsundere_Model extends MonsterInfo{
    public static Id = "GremlinTsundere"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(13, 17);
        else
            return Mathf.RandomRange(12, 15);
    }
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/femaleGremlin/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/femaleGremlin/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/femaleGremlin/skeleton.png"));
    }

    get Name(): string {
        return "持盾地精";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/GremlinTsundere_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class GremlinTsundere_Model : MonsterInfo
    {
        public static string Id = "GremlinTsundere";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(13, 17) : Random.Range(12,15);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/femaleGremlin/skeleton.atlas",
                    "Monsters/theBottom/femaleGremlin/skeleton.json", "Monsters/theBottom/femaleGremlin/skeleton.png");
            }
        }

        public override string Name => "持盾地精";

        public GremlinTsundere_Model(Vector2 pos) : base(pos)
        {
        }
    }
}